ui/pages/Project/components/RnaSeqJunctionOutliersTable.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { RNASEQ_JUNCTION_PADDING } from 'shared/utils/constants'
+import { camelcaseToTitlecase } from 'shared/utils/stringUtils'
+import { GeneSearchLink } from 'shared/components/buttons/SearchResultsLink'
+import DataTable from 'shared/components/table/DataTable'
+import FamilyReads from 'shared/components/panel/family/FamilyReads'
+import { COVERAGE_TYPE, JUNCTION_TYPE } from 'shared/components/panel/family/constants'
+import ShowGeneModal from 'shared/components/buttons/ShowGeneModal'
+import { ButtonLink } from 'shared/components/StyledComponents'
+import { getLocus } from 'shared/components/panel/variants/VariantUtils'
+
+const RNA_SEQ_SPLICE_NUM_FIELDS = ['zScore', 'pValue', 'deltaPsi']
+const RNA_SEQ_SPLICE_DETAIL_FIELDS = ['type', 'readCount', 'rareDiseaseSamplesWithJunction', 'rareDiseaseSamplesTotal']
+
+const RNA_SEQ_SPLICE_COLUMNS = [
+  ...RNA_SEQ_SPLICE_NUM_FIELDS.map(name => (
+    {
+      name,
+      content: camelcaseToTitlecase(name).replace(' ', '-'),
+      format: row => row[name].toPrecision(3),
+    }
+  )),
+  ...RNA_SEQ_SPLICE_DETAIL_FIELDS.map(name => (
+    {
+      name,
+      content: camelcaseToTitlecase(name).replace(' ', '-'),
+    }
+  )),
+]
+
+const getJunctionLocus = (junction) => {
+  const size = junction.end && junction.end - junction.start
+  return getLocus(junction.chrom, junction.start, RNASEQ_JUNCTION_PADDING, size)
+}
+
+class BaseRnaSeqJunctionOutliersTable extends React.PureComponent {
+
+  static propTypes = {
+    reads: PropTypes.object,
+    updateReads: PropTypes.func,
+    data: PropTypes.arrayOf(PropTypes.object),
+    familyGuid: PropTypes.string,
+    tissueType: PropTypes.string,
+  }
+
+  openReads = row => () => {
+    const { updateReads, familyGuid, tissueType } = this.props
+    updateReads(familyGuid, getJunctionLocus(row), [JUNCTION_TYPE, COVERAGE_TYPE], tissueType)
+  }
+
+  render() {
+    const { data, reads, familyGuid } = this.props
+    const junctionColumns = [{
```

Comment:
you should not be creating arrays during render and passing them to a child component, instead you should modify `DataTable` so a call to `format` can take an optional `formatProp` and then you can pass the desired `onClick` function as a top-level prop on the `DataTable`